### ISSUE TITLE
MPT install HOWTO page

### ISSUE THREAD

**Issue by pjritee** [CSSE1001/MyPyTutor#141]
Has anyone been working on this - it would be good to have this all up and going before tomorrow.

**Comment by starsnabove** [comment]
I will put together one at least from a windows (and maybe linux)
perspective. Should be ready later today.

**Comment by sapi** [comment]
I forwarded you what I've already done (had sent it to Ben).

**Comment by pjritee** [comment]
Oh - I'm sorry - is it in the repo?

**Comment by sapi** [comment]
Nah, this is my fault.  I was lazy and just emailed through the changes,
because I didn't want to clone the whole web repo.  They need a bit of work
anyway, so Vincent is probably best placed to fix them up (I emailed them
through to him).

**Comment by starsnabove** [comment]
Updated and currently in place of the old version on the website

**Comment by sapi** [comment]
I'm not seeing it at http://csse1001.uqcloud.net/installing-mpt.html

Are you sure it uploaded correctly?

**Comment by starsnabove** [comment]
Its on the git repo and I did push

**Comment by sapi** [comment]
The website repo or mpt? Not sure how Ben has the former set up (and that's
where it lives atm).

**Comment by starsnabove** [comment]
The website repo. I didnt know how to get it to be on the mpt repo and
reference it from the website repo. As far as I understood it any push to
the repo should make the changes automatically go live. Might need to ask
@athleticus about it.

**Comment by athleticus** [comment]
This is not the case. The website needs to be deployed via rsync (I am the only one who can do that presently). I want to make this a more streamlined process in the future, but it is not very urgent right now.

The reason it hasn't deployed properly is because someone changed the owner/group of the root folder (/opt/local/share/httpd/htdocs) to root/root. It needs to be root/other, else rsync can't create the temporary files it needs in that directory. I can't reset it because I don't have my password to sudo chgrp it.

**Comment by sapi** [comment]
Alex put us on a new zfs share or three, one of which is optimised for
small I/O operations of the kind we're mostly doing.

It's possible that that caused the ownership to change.

I recursively chowned htdocs to www:mypy, so your sync should now succeed.

**Comment by athleticus** [comment]
You also need to chmod g+x the directory.

**Comment by sapi** [comment]
Fixed, recursively.

I always forget the bloody +x!

**Comment by athleticus** [comment]
All g(+x) ;)

**Comment by athleticus** [comment]
You need g+x :/

```
[ben@csse1001 /opt/local/share/httpd/htdocs]$ ls -lah
total 109K
drwxr-xr-x  2 www  mypy   46 Jul 25  2014 _material
drwxr-xr-x 18 www  mypy   41 Mar  3 22:05 .
drwxr-xr-x  7 root root    7 Jun 14  2013 ..
```

**Comment by athleticus** [comment]
It's been a long day.

**Comment by sapi** [comment]
Do you mean g+w? That looks to have g+x

**Comment by athleticus** [comment]
> Do you mean g+w? That looks to have g+x

I edited my last post after I realized it should have been w, not x, and hence why I wrote that it was a long day.

**Comment by athleticus** [comment]
But you're going through email, which explains why you wouldn't have seen the edit.

**Comment by sapi** [comment]
Ah, gotcha. Should be done.

I really need to install my ssh key on my phone...

**Comment by athleticus** [comment]
Thanks. Deployment was successful.

**Comment by pjritee** [comment]
Looks fantastic - the link to the feedback page is broken.
Has anyone checked that the feedback is working - I think I did a couple of weeks ago but that's not to say it's working now.

**Comment by sapi** [comment]
As far as I'm aware, feedback works:

````
[sapi@csse1001 ~/mpt_root/data/feedback]$ pwd -P
/opt/local/share/MyPyTutor/MPT3_CSSE1001/data/feedback
[sapi@csse1001 ~/mpt_root/data/feedback]$ ls
s4206155.0  s4206155.1  uqprobin.0  uqprobin.1
[sapi@csse1001 ~/mpt_root/data/feedback]$ cat uqprobin.1 
{"code": "word1 = input(\"Enter the first word: \")\nword2 = input(\"Enter the second word: \")\n\n# Write your code in the space here:\n\ncombined = word1+word2\n\n\nprint(combined)\n\n\n\n\n\n", "feedback": "testing problem feedback (Input and Output)\n", "subject": "Input and Output"}[sapi@csse1001 ~/mpt_root/data/feedback]$
````

The web page itself doesn't, as it hasn't been updated.  I figured @athleticus or someone else similarly web-inclined could do so at a later date.  For now, we can just read it off the server (I probably should tweak the JSON to have indents, now that I think about it).

**Comment by sapi** [comment]
@starsnabove Really good work with the updated installation instructions; I just read them.

Could you please update them to include information on how to run MPT in the labs?  I'm thinking two things would be useful:
* A note (in the warning style) that if the MPT window briefly appears and then crashes, it probably means they're using Python 2 (eg, atm, double-clicking)
* A link to `run_in_python3.bat` (from [this directory on the server](http://csse1001.uqcloud.net/mpt3/)) for use on the lab machines
